src/node-api-dotnet/index.js

```diff
@@ -11,6 +11,10 @@ const runtimeIdentifier = getRuntimeIdentifier();
 const nativeHostPath = __dirname + `/${runtimeIdentifier}/${assemblyName}.node`;
 const managedHostPath = __dirname + `/${targetFramework}/${assemblyName}.DotNetHost.dll`
 
+// The Node API module may need the require() function at initialization time; passing it as
+// a global is the best (only?) solution, since it cannot be obtained via any `napi_*` function.
+global.require = require;
```

Comment:
It does not look good that we change the global.
Could we instead push the 'require' variable into the module after the its initialization?

Reply:
I don't like it either. I spent some time trying to find a way to get `require` via any `napi_*` function, but nothing worked. We should give that feedback to the Node API working group.

For AOT modules, after initialization can be too late if the module exports some API that references a `Stream`, because setting up the module exports may involve invoking `require()`.

For .NET runtime modules, it is possible to pass in `require` via the host initialization function. I had actually implemented it that way first, but then changed it to be consistent with AOT. But if we expect AOT usage to be much less common then we can at least avoid using the global for non-AOT modules.

We could change AOT modules so that they also have an explicit initialization function. That would mean you can no longer just do a normal `require('MyModule.node')` to load an AOT module. Do you have any thoughts about that?

Reply:
I will merge this as it is now, then follow up later with something a little better.